BusinessLogic/Implementations/RolesAndPermissionsManager.cs

```diff
@@ -14,21 +13,26 @@ public class RolesAndPermissionsManager : IRolesAndPermissionsManager
         private readonly IPermissionsRepository _permissionsRepo;
         private readonly IRolesRepository _rolesRepo;
         private readonly IUserRepository _userRepo;
-        public RolesAndPermissionsManager(IPermissionsRepository permissionsRepo, IRolesRepository rolesRepo, IUserRepository userRepo)
+
+        public RolesAndPermissionsManager(IPermissionsRepository permissionsRepo, IRolesRepository rolesRepo,
+            IUserRepository userRepo)
         {
             _permissionsRepo = permissionsRepo;
             _rolesRepo = rolesRepo;
             _userRepo = userRepo;
         }
+
         public IQueryable<Role> GetAllRoles()
         {
             return _rolesRepo.GetAll();
         }
 
-        public void InsertRole(Role newRole)
+        public Role InsertRole(Role newRole)
         {
-            if (Helpers.HasNullOrEmptyStrings(newRole)) throw new Exception("The New Role To be added Contains null values");
+            if (Helpers.HasNullOrEmptyStrings(newRole))
+                throw new Exception("The New Role To be added Contains null values");
             _rolesRepo.Insert(newRole).Wait();
```

Comment:
Return a task, not an instance. The instance passed as parameter is updated with the id and any other values upon insertion. return a task so that the function is awaitable